torchtitan/models/deepseek_v3/infra/parallelize.py

```diff
@@ -94,7 +93,9 @@ def parallelize_deepseekv3(
         apply_ac(model, job_config.activation_checkpoint)
 
     if job_config.training.compile:
-        raise NotImplementedError("torch.compile is not supported yet for deepseekv3")
+        # NOTE: needed for torch.compile to work with dynamic shapes in token-choice MoE
+        torch._dynamo.config.capture_scalar_outputs = True
```

Comment:
We need to add `import torch` and import `apply_compile`. Will fix in my next PR